admin/vite-build

tohuynh opened the pull request:
<!--
  Thank you for submitting a pull request!

  ⚠️⚠️ Please do the following before submitting: ⚠️⚠️

  - Read the CONTRIBUTING.md guide and make sure you've followed all the steps given.
  - Ensure that the code is up-to-date with the `main` branch.
  - Provide or update documentation for any feature added by your pull request.
  - Provide relevant tests for your feature or bug fix.

  ❗️ Also: ❗️

  Please name your pull request {development-type}/{short-description}.
  For example: feature/read-tiff-files
-->

### Link to Relevant Issue

WIP #134 

### Description of Changes

Build the app and lib using vitejs
#### npm commands for the app
- `npm run start:app` launches the app locally
- `npm run build:app` builds the app in production mode. Generates static site in the `build` dir. See an example [build dir](https://github.com/tohuynh/cdp-frontend/tree/gh-pages) and example [deployment](https://tohuynh.github.io/cdp-frontend/#/).
- `npm run deploy:app` deploys `build` dir (the example app) to forked repo's gh-pages
- `npm run preview:app` launches the app locally, but in production mode at `localhost:5000/cdp-frontend`

#### build lib
`npm run build:lib` builds cdp-frontend as a library to be consumed by various CDP instances by following these [recommendations](https://vitejs.dev/guide/build.html#library-mode). It generates a `dist` dir with 3 files: index.es.js, index.umd.js, and index.css (the assets are inline in the minified CSS file)

#### misc changes
- Removed webpack and related dev dependencies. I had to perform a fresh `npm i` to make it work.
- @BrianL3 I had to change the use of the default cover images a little bit so that they can be imported dynamically in order for them to be included in the final build.

BrianL3 reviewed (approved):
My only comment is that you should maybe update the README/CONTRIBUTION to make sure people know that `npm run start:app` and `npm run build:app` are the ways to get the app up and running. I would like Nick McGruder and other devs new to the project to have a seamless experience getting the app up and running.

tohuynh commented:
> My only comment is that you should maybe update the README/CONTRIBUTION to make sure people know that `npm run start:app` and `npm run build:app` are the ways to get the app up and running. I would like Nick McGruder and other devs new to the project to have a seamless experience getting the app up and running.

Good point

tohuynh commented:
Added some documentation for new npm commands to contributing.md

Added another command `npm run deploy:app` which deploys the example app to your forked repo's gh-pages. Devs can choose to deploy the storybook site or the example app, when making a PR.

hawkticehurst commented on package.json:
```diff
@@ -84,33 +81,22 @@
     "ts-jest": "^26.1.1",
     "ts-node": "^9.1.1",
     "typescript": "^4.5.2",
-    "webpack": "^5.52.0",
```
Have you verified that running the storybook command still works? I ran into an issue at work a couple of months ago where removing webpack resulted in Storybook breaking because it needed webpack as a dependency.

tohuynh replied:
Yes, `npm run storybook` works.

Yes, me too when I tried to remove webpack. I had to run a fresh `npm i` (with webpack removed from package.json) so that  the webpack dependency works out.

hawkticehurst reviewed (approved):
Beyond a question about Storybook still working everything looks great!

JacksonMaxfield reviewed (approved):
Looks good to me! Can someone give me the high level overview of what are the benefits of switching?

tohuynh commented:
> Looks good to me! Can someone give me the high level overview of what are the benefits of switching?

- Build app and lib with the same frontend tooling
- build:app is now a part GitHub CI
- Faster dev server with hot module reload (with cdp icons rendering, snowpack wasn't working before)
- [why Vite?](https://vitejs.dev/guide/why.html)
- [some build optimization](https://vitejs.dev/guide/features.html#build-optimizations)

JacksonMaxfield commented:
Beautiful. Thank you.
